com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainClassHandler.java

```diff
@@ -72,7 +72,7 @@ public void acceptSearchMatch(SearchMatch match) {
                     try {
                         if (method.isMainMethod()) {
                             IResource resource = method.getResource();
-                            if (resource != null) {
+                            if (resource != null && resource.exists()) {
```

Comment:
This will not work. Resource are Java Model object which can be existing even without physical file. You need check the file. See https://github.com/eclipse/eclipse.jdt.ls/commit/064a6a57ff1eb751bf3cc6279909a97e126f501a